extensions/SharkPool/SoundCloud-API.js

```diff
@@ -0,0 +1,363 @@
+
+  let clientID = "8BBZpqUP1KSN4W6YB64xog2PX4Dw98b1";
+  const proxy = "https://corsproxy.io?";
+  const proxy2 = "https://api.codetabs.com/v1/proxy?quest=";
+
+  class SPsoundCloud {
+    getInfo() {
+      return {
+        id: "SPsoundCloud",
+        name: "SoundCloud API",
+        color1: "#ff7600",
+        color2: "#db1b00",
+        color3: "#c02300",
+        menuIconURI,
+        blockIconURI,
+        blocks: [
+          {
+            opcode: "setClient",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set client ID [ID]",
+            arguments: {
+              ID: { type: Scratch.ArgumentType.STRING, defaultValue: "8BBZpqUP1KSN4W6YB64xog2PX4Dw98b1" }
+            }
+          },
+          {
+            opcode: "getClientID",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "get client ID"
+          },
+          "---",
+          {
+            opcode: "extractID",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "get ID of [THING] from url [URL]",
+            arguments: {
+              THING: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "IDS"
+              },
+              URL: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "https://soundcloud.com/"
+              }
+            }
+          },
+          { blockType: Scratch.BlockType.LABEL, text: "Songs" },
+          {
+            opcode: "getTrackAtt",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "get [THING] from track ID [ID]",
+            arguments: {
+              THING: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "TRACKS"
+              },
+              ID: { type: Scratch.ArgumentType.NUMBER, defaultValue: 241049935 }
+            }
+          },
+          {
+            opcode: "getTrackMp3",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "get mp3 of track ID [ID]",
+            arguments: {
+              ID: { type: Scratch.ArgumentType.NUMBER, defaultValue: 241049935 }
+            }
+          },
+          {
+            opcode: "getTrackComment",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "get [NUM2] offset [NUM1] of [TYPE] comments from track ID [ID]",
+            arguments: {
+              TYPE: { type: Scratch.ArgumentType.STRING, menu: "COMMENT" },
```

Comment:
You don't need to define an argument type for a menu

Reply:
Documentation and JeremyGamer say otherwise. 

Garbomuffin can decide

Reply:
_Apparently_ this is wrong but because of some random bug that was only found in PenguinMod so far. I'd close this for now, it's not hurting anyone.